### PR TITLE
Item Alternate Image Features in e-commerce site.

### PR DESCRIPTION
**Pull request by Solufyin**
Hello @nabinhait
Added **Item Alternate Image Features** in e-commerce site.
We request to you closed previous pull request #15015.
As per your changed request #15015 We have developed Item Alternate Image Features.
We keep both features in single doctype. Slideshow and Item alternate image functionality shown in e-commerce site.

**The User can use two alternative choices:**

**1) If user is select slideshow option in slideshow drop down then user is able to see **slideshow functionality** in e-commerce site.**

**2) If user is not select slideshow option in slideshow drop down then a user is able to see **Item Alternate Image** functionality in e-commerce site.**

**We configured images in slideshow item doctype in below snapshot:**

con

**Please find attached video for more information:**

[item_alternate_image.mp4.zip](https://github.com/frappe/erpnext/files/2239954/item_alternate_image.mp4.zip)

Thanks.

### CODE REVIEW

File: erpnext/public/css/website.css [hunk 1]
```diff
@@ -268,3 +268,21 @@
 .bom-spec {
   margin-bottom: 20px;
 }
+/* For item alternate image */
+div.list_all_images img:hover {
```
Comment (netchampfaris) [inline erpnext/public/css/website.css]: Put these styles in website.less

File: erpnext/templates/generators/item.html [hunk 2]
```diff
@@ -3,122 +3,152 @@
...
+                        <div id="item-update-cart" style="display: none;">
+                            <a href="/cart" class='btn btn-sm btn-default'>
+                                <i class='octicon octicon-check'></i>
+                                {{ _("View in Cart") }}</a>
+                        </div>
+                    </div>
+                </div>
+            </div>
+         </div>
+        <div class="row item-website-description" style="margin-top:30px; margin-bottom:20px">
```
Comment (netchampfaris) [inline erpnext/templates/generators/item.html]: Put these inline styles in website.less ?

**Review by netchampfaris — CHANGES_REQUESTED** [review]
I think we should replace the Image Slideshow altogether and set this mode as default.

**Comment by Solufyin** [comment]
Dear @netchampfaris ,

We Improved code as per your suggestion.

**Comment by netchampfaris** [comment]
@Solufyin Hey, as I mentioned in my review, we should remove the old item slideshow and replace it with the new browsing mode you have added. It should become the default mode of browsing multiple images of the item.

**Comment by gmplab** [comment]
i think Solufyin's solution is better. The user can make their own decisions, use the new way or the old way.

**Comment by Solufyin** [comment]
@gmplab, "i think Solufyin's solution is better. The user can make their own decisions, use the new mode or the old mode. You can set the new mode as default."

Yes, The user can make their own choice that's why we keep both functionality slide show and item alternate image in eCommerce site.

**Comment by Solufyin** [comment]
Dear @netchampfaris , 

"@Solufyin Hey, as I mentioned in my review, we should remove the old item slideshow and replace it with the new browsing mode you have added. It should become the default mode of browsing multiple images of the item."

We removed old slideshow and added new item alternate image functionality set as default mode as per your review.

**Comment by netchampfaris** [comment]
item-alternative-image

**Review by netchampfaris — APPROVED** [review]

**Comment by generare** [comment]
This look awesome! All ok with it now?

**Comment by generare** [comment]
Hello again! We were checking the Alternative Images functionality and it will be absolutely nice to have this functionality but it seems to have slightly many steps to input the alternate images if understood correctly. It would be nice if we would not need to jump to a Slideshow interface but rather upload the images on the same item interface page.

We borrowed the images from @Solufyin and edited those a bit to show what we mean.
alternate-images-backend

**Pull request merged**